### ISSUE TITLE
[fabric] Infrared packet portal teleportation not working?

### ISSUE THREAD

**Issue by Sturmlilie** [MightyPirates/TIS-3D#108]
I tested infrared packets flying through a nether portal in 1.14-fabric and couldn't get it to work (on dedicated server), but I'm not sure yet if it's due to the chunk I'm currently not in being immediately unloaded; this will require additional testing.

Since 1.16 completely overhauled how dimensions work and Fabric has a special API just dealing with teleportation at the moment, I might attempt fixing it starting in this version.

**Comment by fnuecke** [comment]
Right, pretty sure it's broken. My test being to have a receiving IR module behind the portal. And it still got the packet.

I also have a pretty good idea why this is... the collision raytrace uses `Raytracing::intersectIgnoringTransparent`. Pretty sure portal blocks are transparent. So this was probably broken for a while now...

Edit: well, that's part of it at least. Teleportation itself is broken, too. Got it kinda working, but for some reason the packets pop out in the inverse direction than they should now. More digging required.

**Comment by unforseenconsequ3nces** [comment]
> 
> 
> Right, pretty sure it's broken. My test being to have a receiving IR module behind the portal. And it still got the packet.
> 
> I also have a pretty good idea why this is... the collision raytrace uses `Raytracing::intersectIgnoringTransparent`. Pretty sure portal blocks are transparent. So this was probably broken for a while now...
> 
> Edit: well, that's part of it at least. Teleportation itself is broken, too. Got it kinda working, but for some reason the packets pop out in the inverse direction than they should now. More digging required.

When is this ever gonna be fixed?

**Comment by UQuark** [comment]
I just thought we may work around portal facing issue.
AFAIK we have no easy way to predict or influence facing rules applied when portal is being generated.
So let's pretend nether portal is some sort of half-transparent quantum mess so that each infrared photon shot into the portal has 50/50 chance to come out facing any of the two sides. Since each packet is made of a bunch of photons, two packets exit in opposite directions on the other side of the portal.
It will also be a satisfactory and predictable behavior for a newcomer even if they think they simply guessed the right side.
